go-controller/pkg/allocator/udn/allocator.go

```diff
@@ -19,6 +20,14 @@ func New(watchFactory factory.NodeWatchFactory, nodeName string) *Allocator {
 	}
 }
 
+func (a *Allocator) AllocateConntrackMark(networkName string) (uint, error) {
+	return a.calculateIDsFromNetwork("ct-mark", networkName, config.Gateway.UserDefinedNetworkConntrackMarkBase, config.Default.MaxUserDefinedNetworks, 1)
+}
+
+func (a *Allocator) AllocateVRFTable(networkName string) (uint, error) {
+	return a.calculateIDsFromNetwork("vrf-table", networkName, config.Gateway.UserDefinedNetworkVRFTableBase, config.Default.MaxUserDefinedNetworks, 1)
```

Comment:
can you help me understand why we need the cardinality which is hardcoded to 1 atm?

Reply:
For masquerade we need a pair of IPs, so the the calculationIDsFromNetwork function got generalized to be able to specify the size of the bucket to reserve using `cardinality` parameter, in the case of VRTable we just need one table from what I see at the enhancement so `cardinality` should be 1.